packages/astro/src/core/build/generate.ts

```diff
@@ -494,128 +513,130 @@ async function generatePath(pathname: string, gopts: GeneratePathOptions, pipeli
 	const manifest = pipeline.getManifest();
 	const { mod, scripts: hoistedScripts, styles: _styles, pageData } = gopts;
 
-	// This adds the page name to the array so it can be shown as part of stats.
-	if (pageData.route.type === 'page') {
-		addPageName(pathname, pipeline.getStaticBuildOptions());
-	}
-
-	pipeline.getEnvironment().logger.debug('build', `Generating: ${pathname}`);
+	for (const route of eachRouteInRouteData(pageData)) {
```

Comment:
The only relevant change is is that the whole logic is inside this loop now, and instead of doing `pageData.route`, we pass the `route` from this loop